### ISSUE TITLE
Beatmap Search Function isn't accurate some times

### ISSUE THREAD

**Issue by rodrigoelicer** [ppy/osu-web#3400]
I already made an issue about this https://github.com/ppy/osu-web/issues/3272 and it already got fixed, but it still has some space for more improvements in my opinion. Let me explain myself.

If I were to find for a single word in order to search for songs related to an artist, currently the search function doesn't discriminate whether that word is included on title or artist or not (probably more fields too), even while being unrrelated. I think the natural behaviour should be to prioritize exact matches first (artist > title > source > mapper) and then try with partial matches, and just then then trying to match with tags, genre, etc.

As an example, I was trying to search for songs related to a group called "TWICE":
https://osu.ppy.sh/beatmapsets?q=%22twice%22
https://osu.ppy.sh/beatmapsets?q=twice
https://osu.ppy.sh/beatmapsets?q=TWICE
And depending whether I search them with quotes, lower case or capital letters, the results seems to vary, which was pretty weird to be honest. Even so, matches like "Lucky Twice" or "Don't Think Twice", being on Artist and Title field respectively appear first, which shouldn't be the behaviour here.

**Comment by rodrigoelicer** [comment]
Was testing a bit more, and seems that if I add my username to the search field 
https://osu.ppy.sh/beatmapsets?q=twice%20iyiyo 
(I'm a mapper and have ranked a couple of maps of this band group), the results looks like a complete mess in my opinion. The very first results are FINE, however, according to the appearance order of the beatmaps, the tags are sometimes coming first than the artist or mapper/creator (being in this case https://osu.ppy.sh/beatmapsets/591331#osu/1514367 this map). Also, if I were a player looking for a mapper and the name of the artist, I'd honestly expect to see first the exact matches, then matches with artist, and then matches with the mapper/creator, but maybe that would be just me.

**Comment by rodrigoelicer** [comment]
Another behaviour I found related to the 'nominations' filter: I was looking for a set with 'escape (TV Size)' as title and filtered it by nominations. (https://osu.ppy.sh/beatmapsets?q=escape%20(TV%20Size)&sort=nominations_desc&s=4) However, you can't see the actual 2 sets with that exact name till theh 18th result.
![image](https://user-images.githubusercontent.com/16049165/42293494-a8913a92-7fa8-11e8-8d21-2235961eb2bc.png)

**Comment by notbakaneko** [comment]
sorting by nominations will orde the results by number of nominations.

**Comment by rodrigoelicer** [comment]
Yeah I know lol, but what happens when you enter something on the search box? Looks like it isn't having any impact on the results.
